From 17a1627f6783ccfb0d98a7ba0a0608b926d5e6fb Mon Sep 17 00:00:00 2001
From: Alon Levy <alevy@redhat.com>
Date: Tue, 29 Jun 2010 19:07:00 +0300
Subject: [PATCH] make opengl optional, disabled by default #482556

The OpenGL renderer isn't really useful right now, its not quite up
to date, its not really faster than software and it only supports a limited
subset of drivers. So, lets disable it for now.

Long term opengl rendering of the 2d part of spice is important if we want
to combine 2d and 3d rendering (say if spice adds opengl support in the
protocol). But until then this is isn't useful for normal use.

Backported from upstream cfc1e95bda0e150b3de225c3572bb1004dad070e
---
 server/configure.ac.shared           |   24 +++++++++++++++++-------
 20 files changed, 134 insertions(+), 50 deletions(-)

diff --git a/server/configure.ac.shared b/server/configure.ac.shared
index eca9943..8c4c876 100644
--- a/server/configure.ac.shared
+++ b/server/configure.ac.shared
@@ -71,6 +71,16 @@ AM_CONDITIONAL(OS_UNIX, test "$os_win32" != "yes")
 AM_CONDITIONAL(OS_LINUX, test "$os_linux" = "yes")

 dnl =========================================================================
+dnl Optional features
+
+have_opengl=no
+AC_ARG_ENABLE(opengl,
+[  --enable-opengl         Enable opengl requirement / support (not recommended)],
+[  have_opengl=yes])
+AM_CONDITIONAL(SUPPORT_GL, test "x$have_opengl" = "xyes")
+
+
+dnl =========================================================================
 dnl Check Dependencies

 SPICE_REQUIRES=""
@@ -139,19 +149,19 @@ SPICE_REQUIRES+=" openssl"
 # AC_SUBST(GL_LIBS)
 # SPICE_REQUIRES+=" gl glu"

-have_gl=yes
-AC_CHECK_LIB(GL, glBlendFunc, GL_LIBS="$GL_LIBS -lGL", have_gl=no)
-AC_CHECK_LIB(GLU, gluSphere, GL_LIBS="$GL_LIBS -lGLU", have_gl=no)

-if test "x$have_gl" = "xno"; then
-   AC_MSG_ERROR([GL libraries not available])
+if test "x$have_opengl" = "xyes"; then
+   AC_CHECK_LIB(GL, glBlendFunc, GL_LIBS="$GL_LIBS -lGL", have_opengl=no)
+   AC_CHECK_LIB(GLU, gluSphere, GL_LIBS="$GL_LIBS -lGLU", have_opengl=no)
+
+   if test "x$have_opengl" = "xno"; then
+      AC_MSG_ERROR([GL libraries not available])
+   fi
 fi

 AC_SUBST(GL_LIBS)
 SPICE_NONPKGCONFIG_LIBS+=" $GL_LIBS"

-
-
 dnl =========================================================================
 dnl configure arguments

-- 
1.7.0.1

